src/App.js

```diff
@@ -10,13 +10,35 @@ class App extends Component {
 
     render() {
         return (
-            <div className="App">
-                <input placeholder="city,country code" onChange={this.onCityChange}/>
-                <button onClick={this.onCitySubscribe}>Subscribe</button>
+            <div>
+                <div className="App">
+                    <input placeholder="city,country code" onChange={this.onCityChange}/>
+                    <button onClick={this.onCitySubscribe}>Subscribe</button>
+                    <button onClick={this.onPrint}>Print</button>
+                </div>
+                    {Object.entries(this.state.subscribedCities).map(([city, data]) => this.renderCityData(city, data))}
             </div>
         );
     }
 
+    onPrint = () => {
+        console.log(this.state.subscribedCities);
+    };
+
+    renderCityData = (city, data) => {
+        return data.list.map( datum => {
+            return (
+                <div>{city}
```

Comment:
Why is this `{city}` left alone outside of a `<span>` or some sort of container?

Reply:
You're also missing a `key` on this `<div>`.
You **don't** need keys on the inner `<div>`s, but you do on this one, as it's the root element in each element item (after the `.map`)
```suggestion
                <div key={city}>{city}
```

Reply:
That `{city}` was being used as a header for the displayed data and np I thought each item had to have its own unique key.

When you say root element are you referring to the fact that it's the parent container of the rendered data?